### PR TITLE
[post] Getting Started with Style Queries

### PR DESCRIPTION
**Pull request by una**
Adds blog post for Style Queries with custom properties landing in M112

### CODE REVIEW

File: site/en/blog/style-queries/index.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  tab: 'result'
+} %}
+
+The above code allows us to apply a chip for `--detail: low-stock` and `--detail: new`, but you may have noticed some redundancy in the code block. Currently, there’s no way to query for just the presence of `--detail` with `@container style(--detail)`, which would allow for better sharing of styles and less repetition. This capability is currently [in discussion](https://github.com/w3c/csswg-drafts/issues/8127) in the working group.
+
+### Weather cards
+
+The previous example used a single custom property with multiple possible values to apply styles. But you can mix it up by using and querying for multiple custom properties too. Take this weather card example:
+
+{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/VCk15tNjkP3Emrrw2Xkl.png", alt="Weather cards demo", width="800", height="255" %}
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:158]:
```suggestion
{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/VCk15tNjkP3Emrrw2Xkl.png", alt="Weather cards demo.", width="800", height="255" %}
```

File: site/en/blog/style-queries/index.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+title: "Getting Started with Style Queries"
+description: >
+  Style queries allow developers to query a parent element's style values using the @container rule. In M112, style queries for CSS custom properties are landing stable. Learn how to get started with them.
+layout: 'layouts/blog-post.njk'
+date: 2023-02-23
+hero: 'image/HodOHWjMnbNw56hvNASHWSgZyAf2/N8vx7PDIeyMNI2wzWnax.jpg'
+alt: >
+  Two-tone pink and blue background with opposite colored purses on each background.
+tags:
+  - chrome-112
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:11]: this is in 111, have confirmed by testing the demo in current Beta. Chrome Status also lists 111.

Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:11]:
```suggestion
  - chrome-111
```

File: site/en/blog/style-queries/index.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+Updating custom property values through JavaScript can be done seamlessly, either while setting up the DOM model (i.e. while building the component in a framework), or updated at any time using `<parentElem>.style.setProperty('--myProperty’, <value>)`. I
+
+Here’s a demo that in a few lines of code, updates the `--theme` of a button, and applies styles using style queries and that custom property (`--theme`)`:
+
+
+{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/mxE8At7l4Ii0O8meY5Zo.mp4", autoplay="true", muted="true", loop="true", alt="Demo cylcing through themes", width="800", height="243" %}
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:210]:
```suggestion
{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/mxE8At7l4Ii0O8meY5Zo.mp4", autoplay="true", muted="true", loop="true", controls="true", alt="Demo cycling through themes", width="800", height="243" %}
```

File: site/en/blog/style-queries/index.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+{% endAside %}
+
+## Separating data from design
+
+In both of these demos, there’s a structural benefit of separating the data layer (DOM that would be rendered on the page) from the styles applied. The styles are written as possible variants that live within the components style, while an endpoint could send the data that it would then use to style the component by. You can use a single value, such as in the first case, updating the `--detail` value, or multiple variables, such as in the second case (setting either `--rainy` or `--cloudy` or `--sunny`. And the best part is that you can combine these values as well, checking for both `--sunny` and `--cloudy` could show a partly cloudy style.
+
+Updating custom property values through JavaScript can be done seamlessly, either while setting up the DOM model (i.e. while building the component in a framework), or updated at any time using `<parentElem>.style.setProperty('--myProperty’, <value>)`. I
+
+Here’s a demo that in a few lines of code, updates the `--theme` of a button, and applies styles using style queries and that custom property (`--theme`)`:
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:207]:
```suggestion
Here’s a demo that in a few lines of code, updates the `--theme` of a button, and applies styles using style queries and that custom property (`--theme`):
```

File: site/en/blog/style-queries/index.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/mxE8At7l4Ii0O8meY5Zo.mp4", autoplay="true", muted="true", loop="true", alt="Demo cylcing through themes", width="800", height="243" %}
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'GRXZLvW',
+  height: 350,
+  tab: 'result'
+} %}
+
+You style the card using style queries, and the JavaScript you use to update the custom property values is:
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:219]:
```suggestion
Style the card using style queries, the JavaScript used to update the custom property values is:
```

File: site/en/blog/style-queries/index.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+```js
+const themePicker = document.querySelector('#theme-picker')
+const btnParent = document.querySelector('.btn-section');
+
+themePicker.addEventListener('input', (e) => {
+  btnParent.style.setProperty('--theme', e.target.value);
+})
+```
+
+As container queries land in browsers and their feature sets grow, expect more opportunities for dynamic, responsive interfaces. As for style queries specifically, there are still a few open issues. One is the implementation of style queries for CSS styles beyond custom properties. This is already a part of the current spec level, but not yet implemented in any brewers. The boolean context evaluation is expected to be added to the current spec level when the outstanding [issue](https://github.com/w3c/csswg-drafts/issues/8127) is resolved, while the [range querying](https://github.com/w3c/csswg-drafts/issues/8376) is planned for the next level of the spec. Exploring style queries with custom properties is now available in Chromium Beta, with more feature additions to come.
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:230]:
```suggestion
The features detailed in this article are just the start. You can expect more things from container queries to help you build dynamic, responsive interfaces. As for style queries specifically, there are still a few open issues. One is the implementation of style queries for CSS styles beyond custom properties. This is already a part of the current spec level, but not yet implemented in any browsers. The boolean context evaluation is expected to be added to the current spec level when the outstanding [issue](https://github.com/w3c/csswg-drafts/issues/8127) is resolved, while [range querying](https://github.com/w3c/csswg-drafts/issues/8376) is planned for the next level of the spec. 
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]
Just a few bits, and I'm pretty sure this landed in 111. It's in the current beta and I can't see anything saying it's being held back, unless you know otherwise.

File: site/en/blog/style-queries/index.md (hunk 2 above)
Comment (una) [reply site/en/blog/style-queries/index.md:11]: Yes, it's in 111! Good catch

**Review by rachelandrew — APPROVED** [review]
Update the date before merging :)

File: site/en/blog/style-queries/index.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
+---
+title: "Getting Started with Style Queries"
+description: >
+  Style queries allow developers to query a parent element's style values using the @container rule. In M112, style queries for CSS custom properties are landing stable. Learn how to get started with them.
```
Comment (rachelandrew) [inline site/en/blog/style-queries/index.md:4]:
```suggestion
  Style queries allow developers to query a parent element's style values using the @container rule. In Chrome 111, style queries for CSS custom properties are landing stable. Learn how to get started with them.
```

**Pull request merged**